apps/wallet/src/ui/app/pages/swap/index.tsx

```diff
@@ -0,0 +1,292 @@
+import { ErrorBoundary } from '_components/error-boundary';
+import Loading from '_components/loading';
+import Overlay from '_components/overlay';
+import { filterAndSortTokenBalances } from '_helpers';
+import {
+	allowedSwapCoinsList,
+	getUSDCurrency,
+	useCoinsReFetchingConfig,
+	useCreateAccount,
+	useGetEstimateSuiToUSDC,
+	useMainnetCoinsMap,
+	useMarketAccountCap,
+	useSortedCoinsByCategories,
+	useSuiBalanceInUSDC,
+} from '_hooks';
+import { useTransactionSummary, useZodForm } from '@mysten/core';
+import { useSuiClientQuery } from '@mysten/dapp-kit';
+import { ArrowDown12, ArrowRight16 } from '@mysten/icons';
+import { type DryRunTransactionBlockResponse } from '@mysten/sui.js/client';
+import { MIST_PER_SUI } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import clsx from 'classnames';
+import { useEffect, useMemo } from 'react';
+import { useWatch, type SubmitHandler } from 'react-hook-form';
+import { Route, Routes, useNavigate, useSearchParams } from 'react-router-dom';
+import { z } from 'zod';
+
+import { AssetData } from './AssetData';
+import { BaseAssets } from './BaseAssets';
+import { GasFeeSection } from './GasFeeSection';
+import { QuoteAssetSection } from './QuoteAssetSection';
+import { initialValues, useCoinTypeData, type FormValues } from './utils';
+
+function getSwapPageAtcText(
+	fromSymbol: string,
+	quoteAssetType: string,
+	coinsMap: Record<string, string>,
+) {
+	const toSymbol =
+		Object.entries(coinsMap).find(([key, value]) => value === quoteAssetType)?.[0] || '';
+
+	return `Swap ${fromSymbol} to ${toSymbol}`;
+}
+
+function SwapPageForm() {
+	const navigate = useNavigate();
+	const [searchParams] = useSearchParams();
+	const activeAccount = useActiveAccount();
+	const signer = useSigner(activeAccount);
+	const activeAccountAddress = activeAccount?.address;
+	const { staleTime, refetchInterval } = useCoinsReFetchingConfig();
+	const { mutate } = useCreateAccount();
+	const {
+		data: marketAccountCapData,
+		refetch: refetchUseMarketAccountCapLoading,
+		isLoading: useMarketAccountCapLoading,
+	} = useMarketAccountCap(activeAccount?.address);
+	const coinsMap = useMainnetCoinsMap();
+
+	const accountCapId = marketAccountCapData?.owner as string;
+
+	const activeCoinType = searchParams.get('type');
+
+	const { isLoading, formattedBalance, coinMetadata } = useCoinTypeData(activeCoinType);
+
+	const { data: coinBalances } = useSuiClientQuery(
+		'getAllBalances',
+		{ owner: activeAccountAddress! },
+		{
+			enabled: !!activeAccountAddress,
+			staleTime,
+			refetchInterval,
+			select: filterAndSortTokenBalances,
+		},
+	);
+
+	const { recognized } = useSortedCoinsByCategories(coinBalances ?? []);
+
+	const formattedTokenBalance = formattedBalance.replace(/,/g, '');
+	const symbol = coinMetadata.data?.symbol ?? '';
+
+	const coinDecimals = coinMetadata.data?.decimals ?? 0;
+	const balanceInMist = new BigNumber(formattedBalance || '0')
```

Comment:
Why this is necessary? I think in line 73 we can get the coin itself which contains the balance in mist already. formattedBalance is calculated from that right?

Reply:
This only shows the raw balance the accountId has in the pool. Different coins will have different conversions to the real values (This is declared when the pool created and cannot really be changed). For SUI, it is this but other coins will have different values.

Reply:
My point is why do we convert it from the formatted value back to the raw value and not taking the raw value straight from the ```	const { isLoading, formattedBalance, coinMetadata } = useCoinTypeData(activeCoinType);
``` at line 73

Reply:
updated code